### PR TITLE
Feature/networks assets api

### PR DESCRIPTION
**Pull request by Psykepro**
**Detailed description**:

**Which issue(s) this PR is needed for**:
Needed for #14

**Special notes for your reviewer**:

**Checklist**
- [ ] Documentation added
- [x] Tests updated

### CODE REVIEW

File: app/router/assets/assets_test.go [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+	assetsResponseHandler := assetsResponse(mocks.MAssetsService)
+	assetsResponseHandler(mocks.MResponseWriter, new(http.Request))
+
+	assert.Nil(t, err)
+	assert.NotNil(t, assetsResponseHandler)
+	assert.NotNil(t, assetsResponseAsBytes)
+}
+
+//
+//func Test_minAmountsResponse_NoAmounts(t *testing.T) {
```
Comment (vectorman1) [inline app/router/assets/assets_test.go:77]: Commented?

File: app/services/assets/assets.go [hunk 2]
```diff
@@ -149,73 +185,155 @@ func (a *Service) fetchHederaFungibleAssetInfo(assetId string, mirrorNode client
...
+	)
+
+	if chainId == constants.HederaNetworkId { // Hedera
+		assetInfo, err = a.fetchHederaNonFungibleAssetInfo(assetAddress, mirrorNode)
+		if err != nil {
+			err = errors.New(fmt.Sprintf("Failed to load Hedera Non-Fungible Asset Info. Error [%v]", err))
+			return assetInfo, assetAddress, err
+		}
+	} else { // EVM
+		re, _ := regexp.Compile(constants.EvmCompatibleAddressPattern)
```
Comment (vectorman1) [inline app/services/assets/assets.go:310]: MustCompile

File: app/services/assets/assets.go [hunk 3]
```diff
@@ -102,7 +112,13 @@ func (a *Service) FungibleAssetInfo(networkId uint64, assetAddressOrId string) (
 	return assetInfo, exist
 }
 
-func (a *Service) fetchEvmFungibleAssetInfo(networkId uint64, assetAddress string, evmTokenClients map[uint64]map[string]client.EVMToken) (assetInfo assetModel.FungibleAssetInfo, err error) {
+func (a *Service) NonFungibleAssetInfo(networkId uint64, assetAddressOrId string) (assetInfo assetModel.NonFungibleAssetInfo, exist bool) {
+	assetInfo, exist = a.nonFungibleAssetInfos[networkId][assetAddressOrId]
```
Comment (vectorman1) [inline app/services/assets/assets.go:116]: exist should be "ok"

File: app/services/assets/assets.go [hunk 4]
```diff
@@ -129,6 +145,26 @@ func (a *Service) fetchEvmFungibleAssetInfo(networkId uint64, assetAddress strin
 	return assetInfo, err
 }
 
+func (a *Service) fetchEvmNonFungibleAssetInfo(networkId uint64, assetAddress string, evmTokenClients map[uint64]map[string]client.EvmNFT) (assetInfo assetModel.NonFungibleAssetInfo, err error) {
+
+	evmTokenClient := evmTokenClients[networkId][assetAddress]
```
Comment (vectorman1) [inline app/services/assets/assets.go:150]: If there is a case where evmTokenClient can't be found we should use ok to check and return appropriate error

File: app/services/assets/assets.go [hunk 5]
```diff
@@ -149,73 +185,155 @@ func (a *Service) fetchHederaFungibleAssetInfo(assetId string, mirrorNode client
...
+
+func (a *Service) loadNonFungibleAssetInfos(networks map[uint64]*parser.Network, mirrorNode client.MirrorNode, evmTokenClients map[uint64]map[string]client.EvmNFT) {
+	a.nonFungibleAssetInfos = make(map[uint64]map[string]assetModel.NonFungibleAssetInfo)
+
+	for nativeChainId, networkInfo := range networks {
+		if len(networkInfo.Tokens.Nft) == 0 {
+			continue
+		}
+
+		if _, exist := a.nonFungibleAssetInfos[nativeChainId]; !exist {
```
Comment (vectorman1) [inline app/services/assets/assets.go:270]: ok

File: bootstrap/clients.go [hunk 6]
```diff
@@ -35,27 +36,29 @@ import (
...
-	HederaNode      client.HederaNode
-	MirrorNode      client.MirrorNode
-	EVMClients      map[uint64]client.EVM
-	CoinGecko       client.Pricing
-	CoinMarketCap   client.Pricing
-	RouterClients   map[uint64]client.DiamondRouter
-	EVMTokenClients map[uint64]map[string]client.EVMToken
+	HederaNode              client.HederaNode
+	MirrorNode              client.MirrorNode
+	EVMClients              map[uint64]client.EVM
```
Comment (vectorman1) [inline bootstrap/clients.go:41]: Evm

File: bootstrap/clients.go [hunk 7]
```diff
@@ -94,13 +97,13 @@ func InitRouterClients(bridgeEVMsCfgs map[uint64]config.BridgeEvm, evmClients ma
...
 }
 
-func InitEVMTokenClients(networks map[uint64]*parser.Network, evmClients map[uint64]client.EVM) map[uint64]map[string]client.EVMToken {
-	tokenClients := make(map[uint64]map[string]client.EVMToken)
+func InitEvmFungibleTokenClients(networks map[uint64]*parser.Network, evmClients map[uint64]client.EVM) map[uint64]map[string]client.EvmFungibleToken {
+	tokenClients := make(map[uint64]map[string]client.EvmFungibleToken)
 	for networkId, network := range networks {
 
 		if networkId != constants.HederaNetworkId {
 			if _, exist := tokenClients[networkId]; !exist {
```
Comment (vectorman1) [inline bootstrap/clients.go:102]: ok

File: bootstrap/clients.go [hunk 8]
```diff
@@ -122,12 +125,56 @@ func InitEVMTokenClients(networks map[uint64]*parser.Network, evmClients map[uin
...
+
+	return tokenClients
+}
+
+func InitEvmNftClients(networks map[uint64]*parser.Network, evmClients map[uint64]client.EVM) map[uint64]map[string]client.EvmNFT {
+	tokenClients := make(map[uint64]map[string]client.EvmNFT)
+	for networkId, network := range networks {
+
+		if networkId != constants.HederaNetworkId {
+			if _, exist := tokenClients[networkId]; !exist {
```
Comment (vectorman1) [inline bootstrap/clients.go:149]: ok

File: bootstrap/clients.go [hunk 9]
```diff
@@ -122,12 +125,56 @@ func InitEVMTokenClients(networks map[uint64]*parser.Network, evmClients map[uin
...
+				tokenClients[networkId][fungibleTokenAddress] = tokenInstance
+			}
+
+			// Wrapped tokens
+			for wrappedNetworkId, wrappedTokenAddress := range tokenInfo.Networks {
+				if wrappedNetworkId == constants.HederaNetworkId {
+					continue
+				}
+
+				if _, exist := tokenClients[wrappedNetworkId]; !exist {
```
Comment (vectorman1) [inline bootstrap/clients.go:171]: ok notation

**Review by vectorman1 — CHANGES_REQUESTED** [review]

**Review by svetlio8 — APPROVED** [review]

File: app/services/assets/assets.go (hunk 4 above)
Comment (Psykepro) [reply app/services/assets/assets.go:150]: There won't be such a case as this function is only used by the bridge config and we initialize the clients by the same config and this function is executed only on start.

**Review by svetlio8 — DISMISSED** [review]

**Review by vectorman1 — APPROVED** [review]

File: app/router/assets/assets.go [hunk 10]
```diff
@@ -0,0 +1,109 @@
...
+
+		// Fungible
+		for _, assetAddress := range fungibleNetworkAssets[networkId] {
+			fungibleAssetInfo, existInfo := assetsService.FungibleAssetInfo(networkId, assetAddress)
+			minAmount, existMinAmount := pricingService.GetTokenPriceInfo(networkId, assetAddress)
+			bridgeTokenInfo := bridgeConfig.Networks[networkId].Tokens.Fungible[assetAddress]
+			if existInfo && existMinAmount {
+				fungibleAssetDetails := fungibleBridgeDetails{
+					FungibleAssetInfo: &fungibleAssetInfo,
+					FeePercentage:     bridgeTokenInfo.FeePercentage,
```
Comment (failfmi) [inline app/router/assets/assets.go:85]: If the token is wrapped, the fee percentage should be inherited from the native token

Comment (failfmi) [inline app/router/assets/assets.go:85]: If the token is wrapped, the fee percentage should be inherited from the native token

Comment (failfmi) [reply app/router/assets/assets.go:85]: Currently, the EVM native token's fee percentages are not returned as well.

Comment (failfmi) [reply app/router/assets/assets.go:85]: Currently, the EVM native token's fee percentages are not returned as well.

**Review by failfmi — APPROVED** [review]

**Pull request merged**